proposals/binary-compat-only.md

```diff
@@ -0,0 +1,123 @@
+# BinaryCompatOnlyAttribute
+
+## Summary
+[summary]: #summary
+
+We introduce a new attribute, `System.BinaryCompatOnlyAttribute`, that causes the type or member to which it is applied to be entirely inaccessible from source code.
```

Comment:
Is it inaccessible from within the current compilation or just referencing compilations?